### DISCUSSION TITLE
Mac版ダウンロードボタン表示問題対策

### DISCUSSION THREAD

**Discussion by CSReviser** [CSReviser/CaptureStream#10] [General]
Mac版ダウンロードボタンが半分しか表示されない問題への対策を意図した修正です。
Macの動作確認する環境がないので、Macユーザーからの情報提供を期待します。

※このバージョンはMac版のみです。

<hr /><em>This discussion was created from the release <a href='https://github.com/CSReviser/CaptureStream/releases/tag/20220416'>Mac版ダウンロードボタン表示問題対策</a>.</em>

**Comment by DaddyJapan** [comment]
[Doc2.docx](https://github.com/CSReviser/CaptureStream/files/8499528/Doc2.docx)
Screenshotを添付しました。うまく動作しています。ありがとうございました。

**Reply by CSReviser** [reply]
ご報告ありがとうございました。
ご協力に感謝いたします。

**Comment by DaddyJapan** [comment]
<img width="584" alt="2022-04-24 11 20 29" src="https://user-images.githubusercontent.com/38952046/164953272-cfc3f6e0-db3b-4091-9a5d-895298769cfc.png">
Macbookで、Githubを使って、Buildを試みましたが、添付のファイルにあるエラーが出てappがbuildできないようです。Windowsの問題のようでした。ご確認できませんか。急ぎません。

**Comment by CSReviser** [comment]
返信が遅れ申し訳ありません。
Macの環境がないので、Macに関しては適切に対応できません。ご理解お願い致します。
尚、リポジトリ「CaptureStream」内のActions（workflows）で動作確認できているのは「c-cpp.yml」（Actions内での表示「C/C++ CI 」）のみです。
Macユーザの皆さんのご協力で動作確認いただいたバイナリ（実行ファイル）は「c-cpp.yml」（Actions内での表示「C/C++ CI 」）を使って作成しています。

**Comment by DaddyJapan** [comment]
ありがとうございました。エラーがでないで終わりましたが、CaptureStream.appが見つかりませんでした。どこのフォルダーに作成されますか。全くの素人ですみません。

**Comment by CSReviser** [comment]
githubの「Actions」の緑のレ丸の右に「C/C++ C」をクリックして開いた画面の下の方にName「CaptureStream」があると思います。
「CaptureStream」をクリックすると「CaptureStream.zip」というZIPファイルがダインロードされ、その中に「CaptureStream.app」があります。
Macの操作方法がわからないので、Macでのダウンロードの方法やZIPファイルの開き方などは、お調べください。
私自身も手探り状態なので、必要な「CaptureStream.app」だけでなく、他のファイルもまとめてZIPにしています。
「CaptureStream.app」だけを取り出して、「CaptureStream.zip」は削除しています。

**Reply by DaddyJapan** [reply]
ありがとうございました。githubからCaptureStream.appをうまく取り出せました。また、ffmpegとStylesheet-mac.qssを前の版からコピーしてきました。それでうまく動作しました。もともと、5/13の問題はありませんでしたので、作成したもので、今週分をダウンロードできました。
今日、アップロードして頂いたものは、dmgファイルでしたが、うまく展開できませんでしたので、何もしていません。エラーも出ずに、終わってしまいます。変更分がffmpegの5.0.1版だけであれば、少なくとも私の環境ではすぐに対応する必要はなさそうです。
明日、時間があれば、ffmpegだけ5.0.1版をダウンロードして試してみることはできそうです。
私の環境:　macOS Monterey Version 12.3.1 MacBook Pro (14-inch, 2021) Chip Apple M1 Pro Memory 16 GB

**Reply by DaddyJapan** [reply]
すみません。先程ダウンロードしたファイルのサイズを確認しましたら、バラバラで、聞いてみると確かに、途中で終わっているのがあります。番組とは関係なく、どの講座でもこの現象が起きるようです。4/16の版に、ffmpeg-5.0.1版を入れても同じ現象でした。

**Reply by DaddyJapan** [reply]
5/15日に作成されたCaptureStream.appをダウンロードして、Stylesheet-mac.qss、ffmpeg-5.0.1を組み合わせて、作ってみましたら、うまく動作しました。dmgファイルがうまく作成できていないだけのようです。ありがとうございました。

**Reply by DaddyJapan** [reply]
いろいろ試して見ました。現在、登録されているファイルでも、きちんと展開できて、動作も確認できました。いろいろお騒がせしてすみません。

**Reply by DaddyJapan** [reply]
[Capture.docx](https://github.com/CSReviser/CaptureStream/files/8717359/Capture.docx)
mp3対応版のdmgファイルを展開すると、期待したようにならなかったので、Actionsの中のCaptureStream.appに、前の版のffmpegなどを入れたものを作って動作させました。mp4を使っているので、それとの比較をしています。

**Reply by CSReviser** [reply]
ご連絡ありがとうございます。
dmgファイルは作成ミスでした。作り直して差し替えました。

m4aファイルとmp3ファイルでファイルサイズが異なるのは現在の仕様のとおりです。
m4aファイルは配信されている音声(コーディック：aac)をそのまま格納しています。（ビットレート:48k）
mp3ファイルはビットレート:64kに再エンコードしています。
m4aと同程度のmp3ファイルをご希望の場合は、拡張子を「op1」を選択することでビットレート:48kのmp3ファイルになりますので、m4aファイルと同程度のファイルサイズのmp3ファイルになると思います。
ビットレートの数値が大きいほど、ファイルサイズは大きくなります。

**Reply by DaddyJapan** [reply]
本日版で動作確認できました。ありがとうございました。
